GitUI/LeftPanel/RepoObjectsTree.cs

```diff
@@ -632,6 +632,11 @@ private void SelectNode(NodeBase node, bool multiple, bool includingDescendants)
 
         private void OnNodeDoubleClick(object sender, TreeNodeMouseClickEventArgs e)
         {
+            if (!AppSettings.DontConfirmBranchCheckout.Value && !MessageBoxes.ConfirmBranchCheckout(this))
+            {
+                return;
+            }
+
```

Comment:
This is too early here - definitely not to be done before HitTest.
And it shall affect branches only.
Hence it needs to be moved to both `LocalBranchNode.Checkout` and `RemoteBranchNode.Checkout`.
The check for `AppSettings.DontConfirmBranchCheckout.Value` can be moved to `MessageBoxes.ConfirmBranchCheckout`.